chapters/annotations.tex

```diff
@@ -68,27 +73,185 @@ \section{Annotations for Documentation}\doublelabel{annotations-for-documentatio
+
+\subsubsection{Variable Replacements}\doublelabel{variable-replacements}
+In most places where text for display is defined the final value of a result
+variable can be embedded by refering to the variable as
+\lstinline!%{inertia1.w}!.  This is similar to the \lstinline!Text! graphical in
+\autoref{text}.
+
+\begin{longtable}[]{|l|l|}
+\hline
+\multicolumn{2}{|l|}{Attributes that can use variable replacements}\\ \hline
+Attribute & Annotation \\ \hline
+\lstinline!title! & \lstinline!Figure! and \lstinline!Plot! \\ \hline
+\lstinline!caption! & \lstinline!Figure! \\ \hline
+\lstinline!legend! & \lstinline!Curve! \\ \hline
+\lstinline!label! & \lstinline!Axis! \\ \hline
+\end{longtable}
+
+Note that expansion to the final value means that expansion is not restricted to
+parameters and constants, so that values to be shown in a caption can be
+determined during simulation.
+
+The percent character is encoded \lstinline!%%!.  Neither \lstinline!%class! nor
+\lstinline!%name! is supported in this context, as this information is expected
+to already be easily accessible (when applicable) in tool-specific ways. (Titles
+making use of \lstinline!%class! or \lstinline!%name! would then only lead to
+ugly duplication of this information.)
+
+\subsubsection{Text Markup in Captions}\doublelabel{text-markup-in-captions}
+In addition to variable replacements, a very restricted form of text markup is
+used for the \lstinline!caption!.
+
+Links take the form \lstinline!%[<text>](<link>)!, where the
+\lstinline![<text>]! part is optional. The \lstinline!<link>! can be in either
+of the following forms:
+\begin{itemize}
+\item A URI, such as
+\lstinline!https://github.com/modelica/ModelicaSpecification! or
+\lstinline!modelica:///Modelica.Blocks!.
+\item A \lstinline!variable:<id>!, where \lstinline!<id>! is a component
+reference such as \lstinline!inertia1.w!.
+\item A \lstinline!plot:<id>!, where \lstinline!<id>! is the identifier of a
+\lstinline!Plot! in the current \lstinline!Figure!.
+\end{itemize}
+
+When \lstinline![<text>]! is omitted, a Modelica tool is free to derive a default based on
+the \lstinline!<link>!.
+
+The styling of the link text and the link action is left for each Modelica tool
+to decide.
+
+\begin{nonnormative}
+For example, \lstinline!%(inertia1.w)! could be displayed as the text
+\lstinline!inertia1.w! formatted with upright monospaced font, and have a pop-up
+menu attached with menu items for plotting the variable, setting its start
+value, or investigating the equation system from which it is solved.  On the
+other hand, \lstinline!%[angular velocity](inertia1.w)! could be formatted in
+the same style as the surrounding text, except some non-intrusive visual clue
+about it being linked.
+\end{nonnormative}
+
+A sequence of one or more newlines (encoded either literally or using the \lstinline!\n!
+escape sequence) means a paragraph break.  (A line break within a paragraph is
+not supported.)
```

Comment:
It should also be stated that newlines before the first paragraph or after the last paragraph are ignored.

Reply:
Addressed by #2581